mycroft/skills/core.py

```diff
@@ -204,6 +205,11 @@ def register_intent(self, intent_parser, handler):
         def receive_handler(message):
             try:
                 handler(message)
+                if len(self.results) > 0:
+                    self.emitter.emit(
+                        Message("results",
+                                metadata=self.results))
+                    self.results.clear()
```

Comment:
I think you would want this to be in a finally block; if handler throws an exception, it would leave the results dict in an unknown state, to be flushed by the next caller.